### PR TITLE
feat: Add multi-line comments

### PR DESCRIPTION
**Pull request by f01dab1e**
# Description

support for multi-line comments: https://doc.rust-lang.org/reference/comments.html

# PR Checklist

- [x] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

**Review by kevaundray — APPROVED** [review]

**Comment by kevaundray** [comment]
@Savio-Sou or @signorecello pinging for visibility

File: crates/noirc_frontend/src/lexer/lexer.rs [hunk 1]
```diff
@@ -335,6 +338,31 @@ impl<'a> Lexer<'a> {
...
+            match ch {
+                '/' if self.peek_char_is('*') => {
+                    self.next_char();
+                    depth += 1;
+                }
+                '*' if self.peek_char_is('/') => {
+                    self.next_char();
+                    depth -= 1;
+
+                    // This block comment is closed, so for a construction like "/* */ */"
```
Comment (jfecher) [inline crates/noirc_frontend/src/lexer/lexer.rs:354]: Unfinished sentence

File: crates/noirc_frontend/src/lexer/lexer.rs [hunk 2]
```diff
@@ -335,6 +338,31 @@ impl<'a> Lexer<'a> {
...
+
+        while let Some(ch) = self.next_char() {
+            match ch {
+                '/' if self.peek_char_is('*') => {
+                    self.next_char();
+                    depth += 1;
+                }
+                '*' if self.peek_char_is('/') => {
+                    self.next_char();
+                    depth -= 1;
```
[the remark this answers is not in the document]
Comment (f01dab1e) [reply crates/noirc_frontend/src/lexer/lexer.rs:352]: #1958

**Comment by critesjosh** [comment]
@f01dab1e follow up question for documentation purposes. In the link to the Rust docs in the PR description, there is no comment labelled "multi-line". Does this PR only add support for BLOCK_COMMENTs that start with `/*` and end with `*/`?

**Comment by f01dab1e** [comment]
yes, I use block comments to indicate multi-line comments. 'doc comments' can be misleading because Noir does not actually support doc comments. while using them will not result in an error, they will simply be ignored.

**Pull request merged**